spec/acceptance/self_service_spec.rb

```diff
@@ -15,17 +15,8 @@
     # Test Confirms all facts are false which is another indicator the class is performing correctly
     describe 'check no self_service fact is false' do
       it 'if idempotent all facts should be true' do
-        expect(host_inventory['facter']['self_service']['S0001']).to eq true
-        expect(host_inventory['facter']['self_service']['S0002']).to eq true
-        expect(host_inventory['facter']['self_service']['S0003']).to eq true
-        expect(host_inventory['facter']['self_service']['S0004']).to eq true
-        expect(host_inventory['facter']['self_service']['S0005']).to eq true
-        expect(host_inventory['facter']['self_service']['S0006']).to eq true
-        expect(host_inventory['facter']['self_service']['S0007']).to eq true
-        expect(host_inventory['facter']['self_service']['S0008']).to eq true
-        expect(host_inventory['facter']['self_service']['S0009']).to eq true
-        expect(host_inventory['facter']['self_service']['S0010']).to eq true
-        expect(host_inventory['facter']['self_service']['S0011']).to eq true
+        expect(host_inventory['facter']['self_service'].size).to eq(11)
+        expect(host_inventory['facter']['self_service'].filter { |_k, v| !v }).to be_empty
```

Comment:
looks good to me, could you explain what this line is doing for my own knowledge

Reply:
Ah yeah instead of adding one row with the same content here, we are just ensuring the hash has 11 elements in it and that none of them are false. We can revert this if that helps make it easier. line 18 above would just need to be increased for each fact we add. The one positive is that spec tests will fail if this is not updated.

 expect(host_inventory['facter']['self_service'].size).to eq(11) # Ensure that there are exactly 11 chunks present in the fact.

expect(host_inventory['facter']['self_service'].filter { |_k, v| !v }).to be_empty # Ensure that none of those are false. We are just doing a filter to get the key and value, then checking the value to make an structure of elements that have the value as false.